security/medusa/l3/registry.c

```diff
@@ -197,182 +158,186 @@ int med_unregister_kclass(struct medusa_kclass_s * ptr)
-	);
+	struct medusa_evtype_s *p;
+
+	med_evtype->name[MEDUSA_EVNAME_MAX-1] = '\0';
+	med_evtype->arg_name[0][MEDUSA_ATTRNAME_MAX-1] = '\0';
+	med_evtype->arg_name[1][MEDUSA_ATTRNAME_MAX-1] = '\0';
+	/* TODO: check whether kclasses are registered, maybe register automatically */
+	med_pr_info("Registering event type %s(%s:%s->%s:%s)\n", med_evtype->name,
+		med_evtype->arg_name[0],med_evtype->arg_kclass[0]->name,
+		med_evtype->arg_name[1],med_evtype->arg_kclass[1]->name);
 	MED_LOCK_W(registry_lock);
-	for (p=evtypes; p; p=p->next)
-		if (!mystrcmp(p->name, ptr->name)) {
+	for (p = evtypes; p; p = p->next)
+		if (strcmp(p->name, med_evtype->name) == 0) {
 			MED_UNLOCK_W(registry_lock);
-			med_pr_err("Error: such event type already exists.\n");
+			med_pr_err("Error: '%s' event type already exists.\n", med_evtype->name);
 			return -1;
 		}
-	ptr->next = evtypes;
-	ptr->bitnr = flags;
+
+	med_evtype->next = evtypes;
+	med_evtype->bitnr = flags;
 
 #define MASK (~(MEDUSA_EVTYPE_TRIGGEREDATOBJECT | MEDUSA_EVTYPE_TRIGGEREDATSUBJECT))
 	if (flags != MEDUSA_EVTYPE_NOTTRIGGERED)
-		for (p=evtypes; p; p ? (p=p->next) : (p=evtypes))
+		for (p = evtypes; p; p ? (p = p->next) : (p = evtypes))
 			if (p->bitnr != MEDUSA_EVTYPE_NOTTRIGGERED &&
-				(p->bitnr & MASK) == (ptr->bitnr & MASK)) {
+				(p->bitnr & MASK) == (med_evtype->bitnr & MASK)) {
 
-				ptr->bitnr++; /* TODO: check for the upper limit! */
+				med_evtype->bitnr++; /* TODO: check for the upper limit! */
 				p = NULL;
 				continue;
 			}
 #undef MASK
-	evtypes = ptr;
+
+	evtypes = med_evtype;
 	if (authserver && authserver->add_evtype)
-		authserver->add_evtype(ptr); /* TODO: some day, check for response */
+		authserver->add_evtype(med_evtype); /* TODO: some day, check for response */
 	MED_UNLOCK_W(registry_lock);
 	return 0;
 }
 
 /**
  * med_unregister_evtype - unregister an event type and notify the authserver.
- * @ptr: pointer to the event type to unregister
- *
+ * @med_evtype: pointer to the event type to unregister
+ * FIXME - this is not used anywhere
  */
-void med_unregister_evtype(struct medusa_evtype_s * ptr)
+void med_unregister_evtype(struct medusa_evtype_s *med_evtype)
 {
-	struct medusa_evtype_s * tmp;
-	med_pr_info("Unregistering event type %s\n", ptr->name);
+	struct medusa_evtype_s *tmp;
+	med_pr_info("Unregistering event type %s\n", med_evtype->name);
 	MED_LOCK_W(registry_lock);
-	if (ptr == evtypes)
-		evtypes = ptr->next;
-	else
-		for (tmp = evtypes; tmp; tmp = tmp->next)
-			if (tmp->next == ptr) {
-				tmp->next = tmp->next->next;
-				if (authserver && authserver->del_evtype)
-					authserver->del_evtype(ptr);
-				break;
-			}
+	if (med_evtype == evtypes)
+		evtypes = med_evtype->next;
+		MED_UNLOCK_W(registry_lock);
+		return;
+
+	for (tmp = evtypes; tmp; tmp = tmp->next) {
+		if (tmp->next == med_evtype) {
+			tmp->next = tmp->next->next;
+			if (authserver && authserver->del_evtype)
+				authserver->del_evtype(med_evtype);
```

Comment:
as in case of add_evtype(), it would be nice have a comment... /* TODO: some day, check for response */
:)

Reply:
As we discussed, for now let just put TODO's aside.